financialaid/views.py

```diff
@@ -224,23 +227,56 @@ class FinancialAidActionView(UpdateAPIView):
     queryset = FinancialAid.objects.all()
 
 
-class GetLearnerPriceForCourseView(APIView):
+class CoursePriceListView(APIView):
     """
-    View for retrieving a leaner's price for a course run
+    View for retrieving a learner's price for course runs in all enrolled programs
+    """
+    authentication_classes = (SessionAuthentication,)
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request, *args, **kwargs):
+        """
+        GET handler
+        """
+        user = request.user
+        program_enrollments = ProgramEnrollment.objects\
+            .select_related('user', 'program')\
+            .filter(user=user, program__live=True).all()
+        formatted_course_prices = []
+        for program_enrollment in program_enrollments:
+            try:
+                response_dict = get_formatted_course_price(program_enrollment)
+                response_dict['program_id'] = program_enrollment.program.id
+                formatted_course_prices.append(response_dict)
+            except ImproperlyConfigured:
+                pass
+        return Response(data=formatted_course_prices)
+
+
+class CoursePriceDetailView(APIView):
+    """
+    View for retrieving a learner's price for a course run
     """
     authentication_classes = (SessionAuthentication, )
     permission_classes = (IsAuthenticated, )
 
     def get(self, request, *args, **kwargs):
         """
-        Get request for GetLearnerPriceForCourseView
+        GET handler
         """
-        learner = request.user
-        program = get_object_or_404(
-            Program,
-            id=self.kwargs["program_id"],
-            live=True
-        )
-        return Response(
-            data=get_course_price_for_learner(learner, program)
+        user = request.user
+        program_enrollment = get_object_or_404(
+            ProgramEnrollment,
+            user=user,
+            program__id=self.kwargs["program_id"],
+            program__live=True
         )
+        try:
+            return Response(
+                data=get_formatted_course_price(program_enrollment)
+            )
+        except ImproperlyConfigured:
```

Comment:
here's my problem with that: if we dont handle the exception and we let the server throw a 500, the response that makes it back to the user will be a stack trace. do we have anything sensitive in there? don't know, really, but i'd rather handle the exception and return our own error code